code/modules/mob/mob.dm

```diff
@@ -647,7 +647,12 @@
 	set category = "OOC"
 
 	if (CONFIG_GET(flag/norespawn))
-		return
+		if(!check_rights_for(usr.client, R_ADMIN))
```

Comment:
```suggestion
		if(!check_rights_for(client, R_ADMIN))
```
We are being called from a mob, so we don't need to go through usr since usr = src.